### PR TITLE
No longer support SUNDIALS versions before 3.0.

### PR DESCRIPTION
**Pull request by bangerth**
We already rule this out in our cmake scripts, so the code here is no longer necessary.

/rebuild

### CODE REVIEW

**Review by tamiko — APPROVED** [review]

**Comment by tamiko** [comment]
```
FAILED: /usr/bin/mpicxx   -DBOOST_NO_AUTO_PTR -DDEBUG -Isource/sundials -Iinclude -I/jenkins/workspace/dealii_PR-11825/include -I/jenkins/workspace/dealii_PR-11825/bundled/taskflow-2.5.0/include -I/jenkins/workspace/dealii_PR-11825/bundled/tbb-2018_U2/include -I/jenkins/workspace/dealii_PR-11825/bundled/boost-1.70.0/include -I/jenkins/workspace/dealii_PR-11825/bundled/umfpack/UMFPACK/Include -I/jenkins/workspace/dealii_PR-11825/bundled/umfpack/AMD/Include -I/jenkins/workspace/dealii_PR-11825/bundled/muparser_v2_2_4/include -isystem /usr/include/mpich -isystem /home/dealii/libs/parmetis-4.0.3/include -isystem /home/dealii/libs/trilinos-release-12-10-1/include -isystem /home/dealii/libs/superlu_dist_5.1.2/include -isystem /home/dealii/libs/petsc-3.7.6/include -isystem /home/dealii/libs/adolc-2.6.4-rc1/include -isystem /home/dealii/libs/assimp-4.1.0/include -isystem /home/dealii/libs/hdf5-1.10.1/include -isystem /home/dealii/libs/oce-OCE-0.18.2/include/oce -isystem /home/dealii/libs/p4est-2.0/FAST/include -isystem /home/dealii/libs/slepc-3.7.3/include -isystem /home/dealii/libs/sundials-3.1.0/include -pedantic -fPIC -Wall -Wextra -Wmissing-braces -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -fopenmp-simd -Werror -std=c++14 -Wno-parentheses -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections -Og -MMD -MT source/sundials/CMakeFiles/obj_sundials_debug.dir/kinsol.cc.o -MF source/sundials/CMakeFiles/obj_sundials_debug.dir/kinsol.cc.o.d -o source/sundials/CMakeFiles/obj_sundials_debug.dir/kinsol.cc.o -c /jenkins/workspace/dealii_PR-11825/source/sundials/kinsol.cc

/jenkins/workspace/dealii_PR-11825/source/sundials/kinsol.cc: In member function ‘unsigned int dealii::SUNDIALS::KINSOL<VectorType>::solve(VectorType&)’:

/jenkins/workspace/dealii_PR-11825/source/sundials/kinsol.cc:278:22: error: ‘struct KINMemRec’ has no member named ‘kin_setupNonNull’

             KIN_mem->kin_setupNonNull = true;

                      ^

ninja: build stopped: subcommand failed.
```

File: source/sundials/kinsol.cc [hunk 1]
```diff
@@ -36,14 +36,10 @@
...
 #    include <sundials/sundials_config.h>
-#    if DEAL_II_SUNDIALS_VERSION_GTE(3, 0, 0)
-#      include <kinsol/kinsol_direct.h>
-#      include <sunlinsol/sunlinsol_dense.h>
-#      include <sunmatrix/sunmatrix_dense.h>
-#    else
-#      include <kinsol/kinsol_dense.h>
-#    endif
+#    include <sunlinsol/sunlinsol_dense.h>
+#    include <sunmatrix/sunmatrix_dense.h>
```
Comment (tamiko) [inline source/sundials/kinsol.cc:42]: @bangerth I am puzzled by the compilation failure. The only difference I can spot is that you changed the order of include files.

Comment (bangerth) [reply source/sundials/kinsol.cc:42]: And I didn't even -- clang-format did that for me. Let me see what happens if I change that back...

**Comment by bangerth** [comment]
I think I figured it out. We have both `DEAL_II_SUNDIALS_VERSION_GTE` and `DEAL_II_SUNDIALS_VERSION_LT`, and I mistook the latter for the former in the place that triggers.

Let's see whether that now works.

**Review by masterleinad — APPROVED** [review]

**Pull request merged**